code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,326 @@
+	. = ..()
+	link_owner.balloon_alert(link_owner, "Essence Link cancelled")
+	link_target.balloon_alert(link_target, "Essence Link cancelled")
+	toggle_link(FALSE)
+	essence_link_action.end_ability()
+	REMOVE_TRAIT(link_owner, TRAIT_ESSENCE_LINKED, TRAIT_STATUS_EFFECT(id))
+	REMOVE_TRAIT(link_target, TRAIT_ESSENCE_LINKED, TRAIT_STATUS_EFFECT(id))
+	UnregisterSignal(link_owner, list(COMSIG_MOB_DEATH, COMSIG_MOVABLE_MOVED, COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, COMSIG_XENOMORPH_HEALED_BY_ABILITY))
+	UnregisterSignal(link_target, list(COMSIG_MOB_DEATH, COMSIG_MOVABLE_MOVED, COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, COMSIG_XENOMORPH_HEALED_BY_ABILITY))
+
+/datum/status_effect/stacking/essence_link/tick()
+	if(stacks < max_stacks && COOLDOWN_CHECK(src, attunement_increase))
+		add_stacks(1)
+	if(!link_target.bruteloss || !current_beam)
+		return
+	if(link_owner.plasma_stored < plasma_cost)
+		if(!COOLDOWN_CHECK(src, plasma_warning))
+			return
+		link_owner.balloon_alert(link_owner, "No plasma for link")
+		link_target.balloon_alert(link_target, "No plasma for link")
+		COOLDOWN_START(src, plasma_warning, plasma_warning_cooldown)
+		return
+	link_target.adjustBruteLoss(-heal_amount)
+	link_owner.use_plasma(plasma_cost)
+
+/// Removes the status effect on death.
+/datum/status_effect/stacking/essence_link/proc/handle_death(datum/source)
+	SIGNAL_HANDLER
+	essence_link_action.end_ability()
+
+/// Toggles the link depending on whether the linked xenos are still in range or not.
+/datum/status_effect/stacking/essence_link/proc/handle_dist(datum/source)
+	SIGNAL_HANDLER
+	var/within_range = get_dist(link_owner, link_target) <= DRONE_ESSENCE_LINK_RANGE
+	if(within_range == was_within_range)
+		return
+	was_within_range = within_range
+	toggle_link(was_within_range)
+	link_owner.balloon_alert(link_owner, was_within_range ? ("Link reestablished") : ("Link faltering"))
+	link_target.balloon_alert(link_target, was_within_range ? ("Link reestablished") : ("Link faltering"))
+
+/// Shares the Resin Jelly buff with the linked xeno.
+/datum/status_effect/stacking/essence_link/proc/share_jelly(datum/source)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/buff_owner
+	var/mob/living/carbon/xenomorph/buff_target
+
+	if(source == link_target)
+		buff_owner = link_target
+		buff_target = link_owner
+	else
+		buff_owner = link_owner
+		buff_target = link_target
+
+	buff_owner.balloon_alert(buff_owner, "Buff shared")
+	buff_target.balloon_alert(buff_target, "Buff shared")
+	buff_target.visible_message(span_notice("[buff_target]'s chitin begins to gleam with an unseemly glow..."), \
+		span_xenonotice("Through the essence link, [buff_owner] has shared their resin jelly with you."))
+	INVOKE_ASYNC(buff_target, /mob/living/carbon/xenomorph.proc/emote, "roar")
+	buff_target.apply_status_effect(STATUS_EFFECT_RESIN_JELLY_COATING)
+
+/// Shares heals with the linked xeno.
+/datum/status_effect/stacking/essence_link/proc/share_heal(datum/source, amount, list/amount_mod)
```

Comment:
This argument list doesn't match the set you're providing in apply_healing's signal call.